### PR TITLE
run inside a docker container

### PR DESCRIPTION
**Pull request by lioramilbaum**
Running the application in a wrapper VM + Refactoring

### CODE REVIEW

File: docker/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+RUN pip install -r requirements.txt \
+  && ansible-galaxy install -r provisioning/playbooks/requirements.yml -p \
+    provisioning/playbooks/roles \
+  && ansible-playbook --connection=local --inventory 127.0.0.1, \
+    --extra-vars "homedir=$TRAVIS_BUILD_DIR" provisioning/playbooks/site.yml
+
+EXPOSE 80
+
+ENTRYPOINT ["./entrypoint.sh", "127.0.0.1", "root", "12345"]
```
Comment (ynixon) [inline docker/Dockerfile]: We have too many places spread around of variables
It should be in one central place.

File: Vagrantfile [hunk 2]
```diff
@@ -60,15 +60,12 @@ Vagrant.configure("2") do |config|
...
-	config.vm.provision :ansible_local, run: 'always', type: :ansible_local do |ansible|
-		ansible.compatibility_mode = "2.0"
-		ansible.galaxy_role_file = 'playbooks/requirements.yml'
-		ansible.galaxy_roles_path = '/vagrant/provisioning/playbooks/roles'
-		ansible.galaxy_command = 'ansible-galaxy install --role-file=%{role_file} --roles-path=%{roles_path}'
-		ansible.provisioning_path = '/vagrant/provisioning'
-		ansible.inventory_path = 'hosts'
-		ansible.playbook = 'playbooks/site.yml'
-	end
+  config.vm.provision "docker" do |d|
```
Comment (ShimiTaNaka) [inline Vagrantfile]: I'm not sure I understand the relation of the docker build during the vagrant, isn't it done outside the VM?

File: docker/Dockerfile (hunk 1 above)
Comment (lioramilbaum) [reply docker/Dockerfile]: I agree. Could you take care of after the PR is merged?

File: docker/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+  && apt-add-repository ppa:ansible/ansible \
+  && apt-get update \
+  && apt-get install -y python3-pip python3-dev ansible libmysqlclient-dev \
+  && cd /usr/local/bin \
+  && ln -s /usr/bin/python3 python \
+  && pip3 install --upgrade pip
+
+COPY . /
+
+RUN pip install -r requirements.txt \
```
[the remark this answers is not in the document]
Comment (lioramilbaum) [reply docker/Dockerfile]: It is an entry point for other to implement the separation. It wasn't happening till now by the others.

File: Vagrantfile (hunk 2 above)
Comment (lioramilbaum) [reply Vagrantfile]: No. This one installs docker inside the VM.

File: docker/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+  && ansible-galaxy install -r provisioning/playbooks/requirements.yml -p \
+    provisioning/playbooks/roles \
+  && ansible-playbook --connection=local --inventory 127.0.0.1, \
+    --extra-vars "homedir=$TRAVIS_BUILD_DIR" provisioning/playbooks/site.yml
```
[the remark this answers is not in the document]
Comment (lioramilbaum) [reply docker/Dockerfile]: It is undefined outside of Travis

[the remark this answers is not in the document]
Comment (lioramilbaum) [reply docker/Dockerfile]: fixed it

**Pull request merged**